### PR TITLE
Add scent eliminator spray

### PR DESCRIPTION
**Pull request by FenixFeather**
#### Summary
Content "Add scent eliminator spray"

#### Purpose of change

#53251

#### Describe the solution

Adds scent eliminator based on [this product](https://www.cabelas.com/shop/en/wildlife-research-center-super-charged-scent-killer-scent-elimination-spray-combo).
- Currently reduces scent completely for normal humans (-500. For context Strong smell trait is 800. Open to suggestions)
- Does not work on non-human scent
- Spawns with other hunting items
- Each application lasts a day

#### Describe alternatives you've considered

- Considered a new scent type, but it seems like these products work based on odor elimination rather than scent masking
- Should `effect_masked_scent` have a decay?

#### Testing

- Spawn scent eliminator
- Apply to self
- Confirm with scent map that scent goes away

### CODE REVIEW

File: data/json/items/ammo.json [hunk 1]
```diff
@@ -1357,5 +1357,18 @@
     "flags": [ "TRADER_AVOID" ],
     "phase": "liquid",
     "ammo_type": "red_pen_ink"
+  },
+  {
+    "type": "AMMO",
+    "id": "scent_eliminator_agent",
+    "name": { "str_sp": "scent eliminator agent" },
+    "symbol": "~",
+    "description": "A liquid that eliminates human scent at the molecular level. Comes in a spray bottle.",
```
Comment (BrettDong) [inline data/json/items/ammo.json:1366]:
```suggestion
    "description": "A liquid that eliminates human scent at the molecular level.  Comes in a spray bottle.",
```

**Comment by Venera3** [comment]
It would be much nicer if instead of a one-off hardcoded effect it would be more flexible - ideally, scent type and intensity defined in the effect definiton, maybe with some scaling for the intensity (though with scent propagation being slow that might be hard to notice).

**Comment by FenixFeather** [comment]
> It would be much nicer if instead of a one-off hardcoded effect it would be more flexible - ideally, scent type and intensity defined in the effect definiton, maybe with some scaling for the intensity (though with scent propagation being slow that might be hard to notice).

The current system is pretty messy. I was also thinking it would be neater for each scent to be its own effect, since it does make sense to be able to have multiple scents at once. Given that we don't use the scent id system that much right now though, I figured maybe it'd be over-engineering to refactor the scent mask system. If that's what we want though, I'm not opposed to tackling it.

**Comment by Shodan14** [comment]
Might be more fun it it eliminated ~90% of a normal human's smell and then slowly scaled down over time. 

This will be even more useful one the precipitation and other systems start interacting with the smell system.

**Comment by Venera3** [comment]
Scent is underused partly because there is no real way to mess with it from the monster side (or wasn't until earlier this experimental cycle, when the monster attack->hidden trait->effect_on_condition solution became an option). If there was a way for effects to change scent type/intensity adding pheromone tags as an ant attack (for example) would be trivial, and they could use effect's built-in removal/resist/block systems instead of a page or two of EOC logic.

**Comment by kevingranade** [comment]
> The current system is pretty messy. I was also thinking it would be neater for each scent to be its own effect, since it does make sense to be able to have multiple scents at once.

Neater yes, but the current scent system is super expensive to process, and each additional scent will multiply the overhead of processing it.
I'm hoping to have some kind of wind map system that charts air flow across the map, and then it becomes much cheaper to determine scent propagation because it's just moving upwind from the monster to see if it encounters a scent source, and that can be shared by different scents.

That aside, the thing I'm actually skeptical of is the product, are there any credible studies of this stuff working? The product page claims it's been found "99% effective" (red flag, marketing term) "at rutgers", but with no reference to the study.

**Comment by FenixFeather** [comment]
> That aside, the thing I'm actually skeptical of is the product, are there any credible studies of this stuff working? The product page claims it's been found "99% effective" (red flag, marketing term) "at rutgers", but with no reference to the study.

I think these products at least somewhat work, as I learned about them via the hunting sim "The Hunt: Call of the Wild". That particular product page is probably overhyped, but the numbers can be adjusted so that it only eliminates x% of your scent rather than all of it.

I did find [this paper](https://bioone.org/journals/Wildlife-Biology/volume-20/issue-4/wlb.00003/Do-available-products-to-mask-human-scent-influence-camera-trap/10.2981/wlb.00003.full) which found that white-tailed deer were captured more often on cameras masked with scent eliminator. The paper's authors speculate that because white-tailed deer suffer from intense hunting pressure from humans, they avoided unmasked cameras, whereas other animals like raccoons, who are not hunted by humans, did not show any aversion to the unmasked cameras. So it seems like there is some evidence that scent elimination products aren't just placebo effect based.

I just set it to remove -800 scent for now since I have no strong opinions on balance, and just wanted an easy way to test that it was working. Easy enough to only do eg -400 to halve your scent, but not eliminate it completely.

**Pull request closed without merging**